### ISSUE TITLE
Missing/ Mispelling a state key crashes the component generation

### ISSUE THREAD

**Issue by JayaKrishnaNamburu** [teleporthq/teleport-code-generators#145]
# Version 0.6

Hey, I was just testing the new UIDL and component generations, here is something i noticed
~~~
              "content": {
                "reference": {
                  "type": "dynamic",
                  "content": {
                    "referenceType": "state",
                    "id": "isVisible"
                  }
                },
~~~

Let us take the above example if i am miss-spelling or not mentioning ``isVisible`` it would be so useful if we skip the generation of that particular statement instead of blocking the whole component generation. Currently if we miss this attribute or wrongly spell that, the component is not being generated as we are trying to find the element in state.

**Comment by JayaKrishnaNamburu** [comment]
And it will be more helpful if we throw the error message more clearly like ``isVisible is missing`` or something like that.

**Comment by alexnm** [comment]
This should be addressed with #41, where we check the UIDL for these inconsistencies. now with the dynamic reference it will be pretty easy to do this.

However, a fallback could work for now, so if you wish to do a check in the generators and throw a custom error message when something is not found in state/prop definitions, feel free to implement this

**Comment by JayaKrishnaNamburu** [comment]
Sure i will work on this 👍

**Comment by alexnm** [comment]
this will be addressed by #41
